### PR TITLE
fix Netlify hosting

### PR DESCRIPTION
**Pull request by micgro42**
Netlify actually needs an index.html in the dist directory. However this
is not added for a library build. So this creates an extra build command
to keep building it as an app that makes still use of the index.html,
link `npm run dev`. This skips the type checking as the build currently
runs on Netlify itself and there we only have a limited amount of build
minutes included. Rather the type checking should be part of the "code
quality" CI job.

This Netlify hosting was broken in #19 / 171d33f

* [ ] Open question: Do we want this to be built in `--mode development`? 
This would allow devtools to work, but might in theory have slight differences
to production?
* [ ] Also: this will need adjustments to the config in Netlify to work. (that is, to the build command)

### CODE REVIEW

File: vite.config.ts [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
 import { resolve } from 'path';
-import { defineConfig } from 'vite';
+import { BuildOptions, defineConfig } from 'vite';
 import vue from '@vitejs/plugin-vue';
 
-// https://vitejs.dev/config/
-export default defineConfig( {
-	build: {
+function getBuildConfig( isAappBuild: bool ): BuildOptions {
+	if ( isAappBuild ) {
```
Comment (lucaswerkmeister) [inline vite.config.ts:8]:
```suggestion
function getBuildConfig( isAppBuild: bool ): BuildOptions {
	if ( isAppBuild ) {
```

File: vite.config.ts [hunk 2]
```diff
@@ -17,7 +22,12 @@ export default defineConfig( {
 		rollupOptions: {
 			external: [ '@vue/compat', 'vue', 'vuex' ],
 		},
-	},
+	};
+}
+
+// https://vitejs.dev/config/
+export default defineConfig( {
+	build: getBuildConfig( !!process.env.BUILD_AS_APP ),
```
Comment (lucaswerkmeister) [inline vite.config.ts:30]: Optional: maybe we should `import { env } from 'process';`?

**Review by lucaswerkmeister — CHANGES_REQUESTED** [review]

File: vite.config.ts (hunk 1 above)
Comment (micgro42) [reply vite.config.ts]: 🤦

Thanks!

**Review by lucaswerkmeister — APPROVED** [review]

File: vite.config.ts (hunk 2 above)
Comment (micgro42) [reply vite.config.ts:30]: I think I prefer `process.env` as it is such an established pattern and it makes it obvious right away that it is coming from the environment variables directly, without needing to scroll up and see where `env` comes from.

Comment (lucaswerkmeister) [reply vite.config.ts:30]: We could still have that with `import process from 'process'` if I’m not mistaken – to me it feels weird how Node has some of these modules that you can `import`/`require` but also they just exist by default for some reason. ([The docs](https://nodejs.org/api/process.html#process) say that the explicit import is recommended, apparently.)

**Review by lucaswerkmeister — APPROVED** [review]

**Pull request merged**